### DISCUSSION TITLE
Python formula is using LibreSSL instead of OpenSSL

### DISCUSSION THREAD

**Discussion by quackerex** [Homebrew/discussions#5153] [Everyday usage] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I got this warning when running python project.
```
NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
```

Is it intended behaviour for the `python@3.11` formula to use macOS `LibreSSL` instead of the installed `openssl@3` dependency for `python@3.11`?

**Comment by Bo98** [comment, marked as answer]
Sounds like you're using Apple's Python?

```console
$ $(brew --prefix)/bin/python3.11 -c 'import ssl; print(ssl.OPENSSL_VERSION)'
OpenSSL 3.2.1 30 Jan 2024

$ /usr/bin/python3  -c 'import ssl; print(ssl.OPENSSL_VERSION)'
LibreSSL 2.8.3
```

**Reply by quackerex** [reply]
Yep you're are correct. I am using `python3 -m venv .venv` to create `venv`. It seems like `python3` no longer link to Homebrew Python...
<img width="241" alt="image" src="https://github.com/Homebrew/discussions/assets/41659980/574d2eaf-9d3a-4ca6-ab7c-19a7ec453e84">

Anyways the warning no longer appear with `venv` created with Homebrew Python.

**Reply by Bo98** [reply]
We changed our default `python3` to be Python 3.12 so that might be why you saw that if you don't have 3.12 installed.

(Though this can be overridden if necessary.)
